### DISCUSSION TITLE
Deprecate swc_ecma_dep_graph?

### DISCUSSION THREAD

**Discussion by dsherret** [swc-project/swc#8289] [General]
`swc_ecma_dep_graph` was initially added to swc in https://github.com/swc-project/swc/pull/1010

There are some changes I'd like to make to this, but to do that, I'm going to fork this code back into Deno's repositories because it might fall out of the scope of swc and we'd like to have full control over this api at this point.

Looking at the [dependents](https://crates.io/crates/swc_ecma_dep_graph/reverse_dependencies), it's only the following crates using this:

- deno_ast
- swc_ecmascript - [0 usages](https://github.com/search?q=repo%3Aswc-project%2Fswc%20analyze_dependencies&type=code)
- depckeck-rs-core - [0 usages](https://github.com/search?q=repo%3Adenbezrukov%2Fdepcheck-rs%20analyze_dependencies&type=code)

So I propose that we just deprecate this code from swc. I can open a PR if you would like?

**Comment by kdy1** [comment]
Yeah, it would be nice. Thank you in advance!
